hoomd/Variant.h

```diff
@@ -374,5 +405,183 @@ class PYBIND11_EXPORT VariantCycle : public Variant
     };
 
 
+/** Power variant
+
+    Variant that goes from m_A -> m_B as x ^ (power)
+*/
+class PYBIND11_EXPORT VariantPower : public Variant
+    {
+    public:
+
+        /** Construct a VariantPower.
+
+            @param A the initial value
+            @param B the finial value
+            @param power the power to approach as
+            @param t_start the first timestep
+            @param t_size the length of the approach
+        */
+        VariantPower(Scalar A, Scalar B, double power,
+                     uint64_t t_start, uint64_t t_size)
+            : m_A(A),
+              m_B(B),
+              m_power(power),
+              m_t_start(t_start),
+              m_t_size(t_size)
+            {
+            m_offset = computeOffset(m_A, m_B);
+            setStartEnd();
+            }
+
+        /// Return the value.
+        Scalar operator()(uint64_t timestep)
+            {
+            if (timestep <= m_t_start)
+                {
+                return m_A;
+                }
+            else if (timestep < m_t_start + m_t_size)
+                {
+                double s = double(timestep - m_t_start) / double(m_t_size);
+                double inv_result =  m_inv_end * s + m_inv_start * (1.0 - s);
+                return pow(inv_result, m_power) - m_offset;
+                }
+            else
+                {
+                return m_B;
+                }
+            }
+
+        /// Set the starting value.
+        void setA(Scalar A)
+            {
+            m_A = A;
+            setInternals();
+            }
+
+        /// Get the starting value.
+        Scalar getA()
+            {
+            return m_A;
+            }
+
+        /// Set the ending value.
+        void setB(Scalar B)
+            {
+            m_B = B;
+            setInternals();
+            }
+
+        /// Get the ending value.
+        Scalar getB()
+            {
+            return m_B;
+            }
+
+        /// Set the approaching power.
+        void setPower(double power)
+            {
+            m_power = power;
+            setStartEnd();
+            }
+
+        /// Get the ending value.
+        Scalar getPower()
+            {
+            return m_power;
+            }
+
+        /// Set the starting time step.
+        void setTStart(uint64_t t_start)
+            {
+            m_t_start = t_start;
+            }
+
+        /// Get the starting time step.
+        uint64_t getTStart()
+            {
+            return m_t_start;
+            }
+
+        /// Set the length of the ramp.
+        void setTSize(uint64_t t_size)
+            {
+            // doubles can only represent integers accuracy up to 2**53.
```

Comment:
Is this relevant because the calculations of the power require you to convert to double, so that sets the limit rather than the unsigned int type?
Also:
```suggestion
            // Doubles can only represent integers accurately up to 2**53.
```

Reply:
Yes, otherwise the conversion would be impossible to convert correctly.